backend/control/control.go

```diff
@@ -34,44 +40,26 @@ type CartographyJobRequest struct {
 	DefaultRegion      string `json:"default_region,omitempty"`
 }
 
-// RuleExecutionLoop tries to execute rules when
-// 1) The ExecuteRules boolean is true
-// 2) The cartography job is not running.
-// When these conditions are true, it also marks ExecuteRules false.
-// And it sets the LastScanCompleted field - right now, for *all* accounts
-func RuleExecutionLoop(postgresDb *gorm.DB, driver neo4j.Driver, ruleDataList []models.RuleData,
-	rulesToExecute []types.Rule) {
+// ExecuteRules attempts to
+//  1. Execute the rules and insert the rule results in postgress DB
+//  2. After rules are executed, the scan_status and last_scan_completed
+//     columns of RULES_RUNNING accounts in AccountDetails table in postgress DB are updated.
+func ExecuteRules(postgresDb *gorm.DB, driver neo4j.Driver, rulesToExecute []types.Rule) {
```

Comment:
Can we call this `ExecuteRulesLoop` and have the stuff inside the for loop after the sleep be put into a function `ExecuteRulesFunc`?

Then at the start of `ExecuteRulesFunc` we can do

```
ExecuteRulesMutex.Lock()
defer ExecuteRulesMutex.Unock()
```

That way we don't need to remember to unlock the mutex every time in the loop